manifests/versionlock.pp

```diff
@@ -0,0 +1,36 @@
+# @summary Locks package from updates.
+#
+# @example Sample usage
+#   zypprepo::versionlock { '0:bash-4.1.2-9.sles12.*':
+#     ensure => present,
+#   }
+#
+# @param ensure
+#   Specifies if versionlock should be `present`, `absent` or `exclude`.
+#
+# @note The resource title must use the format
+#   "%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}".  This can be retrieved via
+#   the command `rpm -q --qf ':%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}'.
+#   Wildcards may be used within token slots, but must not cover seperators,
+#   e.g., 'b*sh-4.1.2-9.*' covers Bash version 4.1.2, revision 9 on all
+#   architectures.
+#
+define zypprepo::versionlock (
+  Enum['present', 'absent'] $ensure = 'present',
+) {
+  require zypprepo::plugin::versionlock
+
+  assert_type(Zypprepo::VersionlockString, $name) |$_expected, $actual | {
+    fail("Package name must be formatted as %{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}, not \'${actual}\'. See Zypprepo::Versionlock documentation for details.")
+  }
+
+  unless $ensure == 'absent' {
```

Comment:
With `ensure => absent`, this peace of code does nothing. This is different of ensuring that it is absent.
Am i missing something ?

Reply:
This was a direct copy of the logic from yum. I think it would be better as if $ensure == 'present'. I will make this change.